dashboard/app/helpers/javalab_files_helper.rb

```diff
@@ -0,0 +1,49 @@
+
+module JavalabFilesHelper
+  # Get all files related to the project at the given channel id as a hash. The hash is in the format
+  # below. All values are StringIO.
+  # {
+  #   "main.json": <main source file for a project>
+  #   "assets": {"asset_name_1": <asset_value>, ...}
+  #   "validation": <all validation code for a project, in json format>
+  #   "maze": <serialized maze if it exists>
```

Comment:
So I was originally thinking that we'd store both main.json and the maze file in the same `sources/` folder. My thinking was that if we eventually have other level content, then it could all just reside in sources. That being said though, the maze file isn't really a source file so I can also see a case for it being in its own directory. I'm fine with this being separate, we'll just have to fetch it accordingly in Javabuilder.

Reply:
I don't have a strong preference here. Would it be better to do the below? That way we have 3 clear categories instead of main.json being an outlier:
```
{
  "sources": {"main.json": "...", "grid.txt": "..."},
  "assets": {...},
  "validation": {...}
}
```

Reply:
Hmm yeah, I think having three categories is clearer and more extensible if we want to add more to sources down the line.

Reply:
👍 will update

Reply:
done!